CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -101,22 +101,9 @@ function Staff:tickDay()
    self:changeAttribute("happiness", 0.05)
   end
 
-  local room = self:getRoom()
-  if room then
-    -- It always makes you happy to see the outdoors (or windows to anywhere)
-    local count = room:countWindows()
-    if room.room_info.id == "staff_room" then -- Pleased another bit
-      count = count * 2
-    end
-    if count > 0 then
-      -- More windows help but in smaller increments
-      self:changeAttribute("happiness", math.round(math.log(count)) / 1000)
-    end
-
-    -- Extra space in the room you are in adds to your happiness
-    local extraspace = (room.width * room.height) / (room.room_info.minimum_size * room.room_info.minimum_size)
-    -- Greater space helps but in smaller increments
-    self:changeAttribute("happiness", math.round(math.log(extraspace)) / 1000)
+  local happiness_factor = self:getRoom() and self:getRoom().happiness_factor
+  if happiness_factor and happiness_factor > 0 then
+    self:changeAttribute("happiness", happiness_factor)
   end
```

Comment:
Why would we need to check happiness is above 0? It would be more open if the possibility for a room to make a staff member unhappy remained.

Reply:
Interesting idea! Fixed both.